packages/java-edition/src/mcfunction/checker/index.ts

```diff
@@ -217,13 +217,31 @@ const itemStack: core.SyncChecker<ItemStackNode> = (node, ctx) => {
 		if (!pair.key) {
 			continue
 		}
-		const key = core.ResourceLocationNode.toString(pair.key, 'full')
-		if (!groupedComponents.has(key)) {
-			groupedComponents.set(key, [])
+		const componentId = core.ResourceLocationNode.toString(pair.key, 'full')
+		if (!groupedComponents.has(componentId)) {
+			groupedComponents.set(componentId, [])
 		}
-		groupedComponents.get(key)!.push(pair.key)
+		groupedComponents.get(componentId)!.push(pair.key)
 		if (pair.value) {
-			nbt.checker.index('minecraft:data_component', key)(pair.value, ctx)
+			if (componentId === 'minecraft:custom_data') {
+				if (pair.value.type === 'nbt:string') {
+					// TODO: Maybe move this to the nbt package
+					const stringNBT = nbt.parser.compound(
+						new core.Source(pair.value.value, pair.value.valueMap),
+						ctx,
```

Comment:
Yeah... it's not ideal that it has to run the parser here. The `#[nbt=mcdoc:custom_item_data[[%parent.id]]] string` mcdoc type should already be parsing the string as an NBT compound.

Reply:
It does, but not in this context. Also, the `nbt` attribute doesn't actually check against the definition yet.

Reply:
The `nbt` attribute does run the checker...

Reply:
(its kinda bugged, there was discussion in discord)